pkg/gardenlet/controller/federatedseed/extensions/controllerinstallation_control.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2020 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package extensions
+
+import (
+	"context"
+	"sync"
+
+	"github.com/gardener/gardener/pkg/api/extensions"
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
+	"github.com/gardener/gardener/pkg/client/kubernetes"
+	kutil "github.com/gardener/gardener/pkg/utils/kubernetes"
+
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/client-go/util/retry"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+type controllerInstallationControl struct {
+	k8sGardenClient kubernetes.Interface
+	seedClient      kubernetes.Interface
+	seedName        string
+	log             *logrus.Entry
+
+	lock              *sync.RWMutex
```

Comment:
Any specific reason, why you are using a pointer to a `sync.RWMutex`?
```suggestion
	lock              sync.RWMutex
```

Reply:
This will result in linting issues:

```
$ make check
> Check
Executing golangci-lint
pkg/gardenlet/controller/federatedseed/extensions/extensions.go:101:46: copylocks: call of s.createControllerInstallationWorkers copies lock value: github.com/gardener/gardener/pkg/gardenlet/controller/federatedseed/extensions.controllerInstallationControl contains sync.RWMutex (govet)
		s.createControllerInstallationWorkers(ctx, s.controllerInstallationControl)
		                                           ^
pkg/gardenlet/controller/federatedseed/extensions/extensions.go:112:87: copylocks: createControllerInstallationWorkers passes lock by value: github.com/gardener/gardener/pkg/gardenlet/controller/federatedseed/extensions.controllerInstallationControl contains sync.RWMutex (govet)
func (s *Controller) createControllerInstallationWorkers(ctx context.Context, control controllerInstallationControl) {
                                                                                      ^
make: *** [check] Error 1
```